cmd/create/idp/cmd.go

```diff
@@ -393,6 +414,9 @@ func run(cmd *cobra.Command, _ []string) {
 		idpBuilder, err = buildGitlabIdp(cmd, cluster, idpName)
 	case "google":
 		idpBuilder, err = buildGoogleIdp(cmd, cluster, idpName)
+	case "htpasswd":
+		createHTPasswdIDP(cmd, cluster, clusterKey, idpName, ocmClient)
```

Comment:
This function calls `doCreateIDP`, but so does this file after the `switch` statement. Would it make sense to follow the same pattern as the rest of the IDPs, where you just call something like `buildHtpasswdIdp` that returns the `idpBuilder`?

Reply:
I wanted to hijack the flow like that because unlike the other IDPs you can't just build the IDP, create it and finish. 

You may need to create the new IDP and then go through the interactive `shouldAddAnotherUser()` loop to post more users to the IDP's user collection. 

Or, the HTPasswd IDP might already exist, if the user already called `rosa create admin` for example. in that case on `rosa create idp --type="htpasswd"` we don't build the IDP at all, we just add the new users to the existing IDP. 

I hope this makes sense.

Reply:
Yes, this makes sense. But it seems like `doCreateIDP` is being called twice (once inside the `createHTPasswdIDP` and another time after the `switch`).

Reply:
That's exactly why there's a return statement right after that. The second call to `doCreate` isn't reached.

Reply:
:facepalm: Of course!! thanks :)

I think you can change that `return` to be `os.Exit(0)`